File: Исходники/Глава 2. Часть 1/Динамическое программирование1/Динамическое программирование1/Табличный метод/6. Числа Фибоначчи. Формула/C++/Олимпиадный/fib_formula.cpp
#include <bits/stdc++.h>
#include <cassert>

using namespace std;


int fib_formula(int n) {
    if (n < 0) {
        return -1;
    }
    double root_f = std::sqrt(5);
    double fie = (1 + root_f) / 2;
    double den = std::pow(fie, n) - std::pow(-fie, -n);
    double num = 2 * fie - 1;
    double fib = den / num;
    return static_cast<int>(fib);
}

File: Исходники/Глава 2. Часть 1/Динамическое программирование1/Динамическое программирование1/Табличный метод/6. Числа Фибоначчи. Формула/Python/fib_formula.py
def fib_formula(n):
    if n < 0:
        return -1
    root_five = sqrt(5)
    f = (1 + root_five)/2
    den = f**n - (- f)**(-n)
    num = 2 * f - 1
    fib = den / num
    return int(fib)